auth.js

```diff
@@ -0,0 +1,137 @@
+const email = document.querySelector("#email");
+const password = document.querySelector("#password");
+const passwordConfirm = document.querySelector("#password_confirm");
+const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+const signUpButton = document.querySelector("#signUpButton");
+const loginButton = document.querySelector("#loginButton");
+signUpButton ? signUpButton.setAttribute("disabled", true) : loginButton.setAttribute("disabled", true);
+
+const passwordVisibleButtons = document.querySelectorAll("#passwordVisibleButton");
+
+const checkEmailData = (e) => {
+  const emailElementParent = e.target.parentNode;
+  const emailErrorMessegeEmpty = "이메일을 입력해 주세요.";
+  const emailErrorMessegeFormat = "잘못된 이메일 형식입니다.";
+
+  let emailErrorMsgElement = document.createElement("p");
+  emailErrorMsgElement.className = "error_msg";
+
+  const emailExistingErrorMsg = emailElementParent.querySelector(".error_msg");
+  if (emailExistingErrorMsg) {
+    emailExistingErrorMsg.remove();
+  }
+
+  if (email.value === "") {
+    email.classList.add("error");
+    emailErrorMsgElement.textContent = emailErrorMessegeEmpty;
+    emailElementParent.append(emailErrorMsgElement);
+  } else if (!emailRegex.test(email.value)) {
+    email.classList.add("error");
+    emailErrorMsgElement.textContent = emailErrorMessegeFormat;
+    emailElementParent.append(emailErrorMsgElement);
+  } else {
+    email.classList.remove("error");
+    const emailExistingError = emailElementParent.querySelector(".error_msg");
+    if (emailExistingError) emailExistingError.remove();
+  }
+
+  submitButtonVisible();
+};
+
+const checkPasswordData = (e) => {
```

Comment:
💊 제안
하나의 함수가 하나의 역할을 하게 코드를 짜는것이 좋습니다.
지금의 경우 errorMessage가 있으면 해당 요소를 지워주는 일을 하고 그후 input value에 따라 적절한 에러 메시지를 노출하는 일을 하고 있습니다. 이렇게 되면 코드가 길어지고 가독성이 떨어지며 재사용성이 낮아집니다.
가능하면 역할별로 분리해주세요~

```suggestion
const checkPasswordData = (e) => {
  clearErrMsg();
  toggleErrorMsg();
  submitButtonVisible();
}
```

위처럼 분리해보시고 email, paaword등 추상화를 통해 공통으로 사용하게 만드는 방법도 고민해보시면 더 좋을 것 같습니다!